src/core/ruby_version.rs

```diff
@@ -277,17 +285,29 @@ macro_rules! get_stack_trace(
             ruby_global_symbols_address_location: Option<usize>,
             source: &T,
             pid: Pid,
-        ) -> Result<StackTrace, MemoryCopyError> {
+            on_cpu: bool,
+        ) -> Result<Option<StackTrace>, MemoryCopyError> {
             let thread: $thread_type = get_execution_context(ruby_current_thread_address_location, ruby_vm_address_location, source)
                 .context(ruby_current_thread_address_location)?;
 
+            let thread_status = get_thread_status(&thread, source)?;
```

Comment:
Tiny optimization - these 2 calls require reading into the remote's memory, so you can avoid them unless actually needed (don't need the status if not `on_cpu`, and don't need `thread_id` if the `on_cpu` check doesn't pass)